flink-table-store-core/src/main/java/org/apache/flink/table/store/file/schema/SchemaManager.java

```diff
@@ -123,6 +123,40 @@ public TableSchema createTable(Schema schema) throws Exception {
             Map<String, String> options = schema.options();
             int highestFieldId = RowType.currentHighestFieldId(fields);
 
+            List<String> columnNames =
+                    schema.fields().stream().map(DataField::name).collect(Collectors.toList());
+            if (options.containsKey(CoreOptions.PRIMARY_KEY.key())) {
+                if (!primaryKeys.isEmpty()) {
```

Comment:
The validation should be added into `SchemaValidation#validateTableSchema`.

Reply:
I tried to move the validation to `SchemaValidation#validateTableSchema`, but I found that the main function of the method is to validate , but we need to modify the values of partition and primary-key, so I think `SchemaManager#createTable` is more appropriate, what do you think about？